lib/web_ui/lib/src/engine/platform_dispatcher.dart

```diff
@@ -971,6 +1028,15 @@ class EnginePlatformDispatcher extends ui.PlatformDispatcher {
     _fontSizeObserver = null;
   }
 
+  /// Updates [appLifecycleState] and invokes [_setAppLifecycleState] if
+  /// [appLifecycleState] changed.
+  void _updateAppLifecycleState(ui.AppLifecycleState value) {
+    if (value != appLifecycleState) {
+      appLifecycleState = value;
+      _setAppLifecycleState(value);
+    }
+  }
```

Comment:
Instead of having this loose function here, please modify the `AppLifecycleStateSupport` class so its instance requires a "broadcastLifecycleState" function when it's created, then pass the `_setAppLifecycleState` function when creating the `instance`.